Source/WebCore/platform/graphics/FlatDynamicRangeLimit.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/ArgumentCoder.h>
+
+namespace WTF {
+class TextStream;
+}
+
+namespace WebCore {
+
+namespace Style {
+struct DynamicRangeLimit;
+}
+
+enum class ApplicableDynamicRangeLimit : uint8_t {
+    Standard,
+    ConstrainedHigh,
+    NoLimit
+};
+
+class FlatDynamicRangeLimit {
```

Comment:
How about `GraphicsDynamicRangeLimit`? We needed a dynamic range limit for the graphics objects (images and videos) not for every possible HTML element. This limit has to be just a float which will be passed to the system frameworks. Also the other `DynamicRangeLimit` is called `Style::DynamicRangeLimit`.

Reply:
Can't we also include `ApplicableDynamicRangeLimit` as sub type of `GraphicsDynamicRangeLimit` say `Limit` or `Type` or something similar. An enum can be used like: `GraphicsDynamicRangeLimit::Limit::NoLimit`

Reply:
Happy to rename to `GraphicsDynamicRangeLimit`.
However eventually this may apply to every HTML element! (There's an upcoming `css-color-hdr` spec.)

As discussed I will remove `ApplicableDynamicRangeLimit`, it's not really needed here.

Reply:
"Flat" doesn't work for me. How about `DynamicRangeLimitValue`?

Reply:
I've actually renamed it to `PlatformDynamicRangeLimit` (before I saw Simon's comment), as I saw kind-of similar/related classes, like `PlatformColorSpace`, `PlatformLayer`, etc.

Let me know if you'd prefer `DynamicRangeLimitValue`.